adc/read_vsys/power_status.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2023 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef POWER_STATUS_H
+#define POWER_STATUS_H
+
+/*!
+ * \brief Get batttery status
+ *
+ * Returns whether battery powered
+ * \note On Pico W must have called cyw43_arch_init
+ *
+ * \param battery_powered True if powered by battery, False if powered by USB or another means
+ * \return Zero if the battery status can be determined, an error code otherwise \see pico_error_codes
+ */
+
+int power_battery(bool *battery_powered);
```

Comment:
Sorry to nitpick, but `power_battery` feels like a very weird function-name? Perhaps `power_is_battery` or `power_source` or `power_status` or something? :shrug: 
(as always, please take my comments with a large pinch of salt!)